contracts/collections/sg721-base/src/upgrades/v0_17.rs

```diff
@@ -0,0 +1,28 @@
+use cosmwasm_std::{CustomMsg, DepsMut, Response};
+use cw721_base_016 as v16;
+use serde::{de::DeserializeOwned, Serialize};
+use sg_std::Response as StargazeResponse;
+
+use crate::ContractError;
+
+pub fn migrate<T, C, E, Q>(deps: DepsMut) -> Result<StargazeResponse, ContractError>
+where
+    T: Serialize + DeserializeOwned + Clone,
+    Q: CustomMsg,
+    E: CustomMsg,
+{
+    // remove old minter info
+    let tract16 = v16::Cw721Contract::<T, C, E, Q>::default();
+    let minter = tract16.minter.load(deps.storage)?;
+    tract16.minter.remove(deps.storage);
+
+    // save new ownership info
+    let ownership = cw_ownable::initialize_owner(deps.storage, deps.api, Some(minter.as_str()))?;
+
+    Ok(Response::new()
+        .add_attribute("action", "migrate")
+        .add_attribute("from_version", "0.16.0")
```

Comment:
Hardcoding versions make upgrading cumbersome. Should be read from env vars.

Reply:
As I suggested in earlier comments, `v0_17.rs` contains the specific upgrade logics for 0.16 => 0.17 and not any other migrations, therefore imo it's reasonable to hardcode it here.